src/visit.ts

```diff
@@ -563,6 +668,94 @@ export function visitEachChild<T extends FunctionlessNode>(
     } else {
       return new ArrayBinding(bindings as ArrayBinding["bindings"]) as T;
     }
+  } else if (isLabelledStmt(node)) {
+    const stmt = visitor(node.stmt);
+    ensure(stmt, isStmt, "LabelledStmt's stmt must be a Stmt");
+    return new LabelledStmt(node.label, stmt) as T;
+  } else if (isSwitchStmt(node)) {
+    const clauses = node.clauses.flatMap((clause) => {
+      const updatedClause = visitor(clause);
+      if (Array.isArray(updatedClause)) {
+        ensureItemOf(
+          updatedClause,
+          isSwitchClause,
+          "must be a CaseClause or DefaultClause"
+        );
+        return updatedClause;
+      } else {
+        ensure(
+          updatedClause,
+          isSwitchClause,
+          "must be a CaseClause or DefaultClause"
+        );
+        return [updatedClause];
+      }
+    });
```

Comment:
I've seen this pattern multiple times now, can be abstract it?

Reply:
Replaced with `.flatMap` and a helper everywhere